### ISSUE TITLE
Spacing issues in lists

### ISSUE THREAD

**Issue by jackbentley** [pradyunsg/furo#97]
**Describe the bug**
List spacing is not consistent depending on list item content.

**To Reproduce**
```html
<ol class="simple">
    <li>
        <p>Numbered lists are not complicated.</p>
        <p>Numbered lists are not complicated.</p>
    </li>
    <li>
        <p>They do exactly what you think they do.</p>
    </li>
    <li>
        <p>Kind of...</p>
    </li>
</ol>
```

**Expected behavior**
Spacing between list items should be consistent regardless of content. Spacing inside the list item should not be affected by it being in a list.

**Screenshots**
<img width="337" alt="Screenshot 2021-03-09 at 19 51 37" src="https://user-images.githubusercontent.com/9383439/110529776-8faf7580-8111-11eb-984d-ea622e479d04.png">

**Additional context**
Worth noting that the current stylesheets expect a `<p>` element to be inside the `<li>` element. This doesn't seem to always be the case, as can be seen on the [psycopg3 docs](https://www.psycopg.org/psycopg3/docs/). The items under "Documentation" are not inside `<p>` elements.

<img width="369" alt="Screenshot 2021-03-09 at 19 58 01" src="https://user-images.githubusercontent.com/9383439/110529992-d0a78a00-8111-11eb-8fc1-6e6e808c2b82.png">

**Comment by pradyunsg** [comment]
Thanks for filing this issue!

> To Reproduce

Could you share the reStructuredText that generates this output?

**Comment by jackbentley** [comment]
@pradyunsg Sure, here you go 😄 

```rst
1. Numbered lists are not complicated.

   Numbered lists are not complicated.
2. They do exactly what you think they do.
3. Kind of...

Below reproduces the screenshot from psycopg3 docs

- Strict Strong Typing
- asynchronous support
- server-side parameters binding
- binary communication
- a great integration of the COPY support
- direct access to the libpq functionalities

.. toctree::
    :maxdepth: 1

    index
    glossary

```

**Comment by pradyunsg** [comment]
Ah, yes. These are all 3 different types of lists, and they have different spacing as a consequence of how I decided to treat them differently. This was informed by how I've seen them being used in other themes and actual documentation sites, that I took inspiration from. :)

- The first one is a "compound" list (a list with at least 1 bullet containing more than 1 paragraph). Looking around, every theme handles them differently.
- The second one is a "simple" list, where most themes leave no margin between the elements, which I found to be sub-optimal for legibility and visual separation.
- The third one is a "toctree" list, which is basically a simple list with each item being an `<a>` tag instead of a `<p>` tag. These have the visual separation (thanks to how the links are stylised) and having a good density here is useful for including more of the tree on the page (especially on mobile devices).

Is there a reason that this difference in how these lists are spaced is a concern? I'll relabel this, since I don't think this is a bug (and hence is a wontfix) but I'm open to having a longer form discussion about these things, if you feel strongly. :)

<details><summary>Screenshot for future reference</summary>

![Screenshot 2021-03-20 at 1 19 52 PM](https://user-images.githubusercontent.com/3275593/111870862-f6892600-897e-11eb-8744-54f9e4a61a41.png)

</details>
